sqlglot/transforms.py

```diff
@@ -80,4 +80,24 @@ def _transform(self, *args, **kwargs):
     return _transform
 
 
+def add_or_dpipe(expression):
+    if isinstance(expression, exp.Add) and expression.type in (
+        exp.DataType.Type.VARCHAR,
+        exp.DataType.Type.NVARCHAR,
+        exp.DataType.Type.CHAR,
+        exp.DataType.Type.NCHAR,
+        exp.DataType.Type.TEXT,
+    ):
+        expression = exp.DPipe(this=expression.this, expression=expression.expression)
+    return expression
+
+
+def transform_add(self, e):
+    expr = add_or_dpipe(e)
+    delegate_func = f"{expr.key}_sql"
+    to_sql = delegate(delegate_func)
+    return to_sql(self, e)
+
+
 UNALIAS_GROUP = {exp.Group: preprocess([unalias_group], delegate("group_sql"))}
+ADD_TO_DPIPE = {exp.Add: transform_add}
```

Comment:
no one really uses types yet so i don't think this should be a mandatory transform,  let's delete ADD_TO_DIPE and transform_add

and then in your own code, you can call query.transform(add_or_pipe)

Reply:
another idea is to put this in the annotator as a post annotation transform